src/main/java/Parser.java

```diff
@@ -0,0 +1,43 @@
+public class Parser {
+
+    TaskList taskList;
+    Ui ui;
+
+    public Parser(TaskList taskList, Ui ui) {
+        this.taskList = taskList;
+        this.ui = ui;
+    }
+
+    public boolean parse(String input) throws DukeException {
```

Comment:
Would it be better if the input were to be split into two types? One that adds tasks, and the other type does not. It would reduce the total line count of the function, making it easier to read.